sirepo/pkcli/jupyterhublogin.py

```diff
@@ -58,9 +60,21 @@ def maybe_create_sirepo_user(qcall, email, display_name):
         with qcall.auth.logged_in_user_set(u, method=sirepo.auth.METHOD_EMAIL):
             n = sirepo.sim_api.jupyterhublogin.create_user(
                 qcall=qcall,
-                check_dir=True,
             )
             qcall.auth_db.model("UserRole").add_roles(
                 roles=[sirepo.auth_role.for_sim_type("jupyterhublogin")],
             )
         return PKDict(email=email, jupyterhub_user_name=n)
+
+
+def old_user_dirs():
+    """Outputs commands to delete dirs for Jupyter users that are not in database."""
+    with sirepo.quest.start() as qcall:
+        m = ""
+        r = qcall.auth_db.model("JupyterhubUser").search_all_for_column("user_name")
```

Comment:
Convert this to a frozenset, because it is inside a loop, and there are NxM checks here otherwise. Indeed, I think search_all_for_column could just return a frozenset. I noticed auth.get_uids calls it, but that function is never called. search_all_for_column is unsorted anyway so it doesn't need to be a list, and it's used in contexts where it is better to be a set. There is one case (pkcli.roles.lists) it should be converted to a list, but that's the atypical case.

Reply:
[c4d59a6](https://github.com/radiasoft/sirepo/pull/7027/commits/c4d59a6548a384c05945af051a95c0d1ecb498ea)

Reply:
It will need to be converted to a list also for authState. (exception=Object of type frozenset is not JSON serializable) In my opinion this change overcomplicates things.

Reply:
OK. Don't change search_all, but do use a frozenset to test the condition. This speeds the inside test to a hash lookup.

Reply:
[9fc0efd](https://github.com/radiasoft/sirepo/pull/7027/commits/9fc0efd75cc4f7666288b72685a967ed6f474cfe)